tests/test_babis.py

```diff
@@ -9,21 +9,79 @@
 """
 
 import pytest
+from mock import ANY, Mock, call, patch
+from six.moves.urllib.error import URLError
 
 
-from babis import babis
+from babis import decorator
 
 
 class TestBabis(object):
+    def test_before_and_after(self):
+        first_request = Mock()
+        second_request = Mock()
+        third_request = Mock()
 
-    @classmethod
-    def setup_class(cls):
-        pass
+        with patch('babis.babis.Request') as Request:
+            Request.side_effect = [
+                first_request,
+                third_request
+            ]
+            with patch('babis.babis.urlopen') as urlopen:
+                def wrapped():
+                    urlopen(second_request)
+                decorator(ping_before='foo', ping_after='bar')(wrapped)()
 
-    def test_something(self):
-        pass
+        urlopen.assert_has_calls([
+            call(first_request), call(second_request), call(third_request)])
 
-    @classmethod
-    def teardown_class(cls):
-        pass
+    def test_ping_urls_string(self):
+        dec = decorator()
+        dec.ping_url = Mock()
```

Comment:
Instead of patching the actual ``dec.ping_url`` attribute, I would prefer to patch ``babis.babis.ping_url``.